src/controllers/pages/student.js

```diff
@@ -47,6 +47,19 @@ export default class StudentController{
         });
     }
 
+    static checkInModel(id, modelAddress, quantity){
+        return checkInModel(id, modelAddress, quantity).then(data => {
+            return searchStudent(id).then(student => {
+                Dispatcher.handleAction('MODEL_CHECKIN_SUCCESS', {
+                    modelAddress: data.modelAddress,
+                    modelName: data.modelName,
+                    quantity: data.quantity
+                });
+                Dispatcher.handleAction("STUDENT_FOUND", student);
+            });
+        });
```

Comment:
Need a .catch at the end of this promise for when the server fails to do the checkin.